Sources/MongoSwift/BSON/BSONEncoder.swift

```diff
@@ -169,7 +169,17 @@ public class BSONEncoder {
             )
         }
 
-        return dict.toDocument()
+        guard let doc = try? dict.toDocument() else {
```

Comment:
same point as above. one error you might get here is that the resulting document would be too large to be valid BSON. we should use try/catch and propagate that error message to the user.